pkg/cloud/services/networking/network.go

```diff
@@ -383,3 +383,15 @@ func getSubnetName(clusterName string) string {
 func getNetworkName(clusterName string) string {
 	return fmt.Sprintf("%s-cluster-%s", networkPrefix, clusterName)
 }
+
+// ConvertOpenStackSubnetToCAPOSubnet converts an OpenStack subnet to a capo subnet and adds to a slice.
+// It returns the slice with the converted subnet.
+func (s *Service) ConvertOpenStackSubnetToCAPOSubnet(subnets []infrav1.Subnet, filteredSubnet *subnets.Subnet) []infrav1.Subnet {
```

Comment:
In case you want to add unit tests:

```golang

(...)
"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
(...)

func Test_ConvertOpenStackSubnetToCAPOSubnet(t *testing.T) {
	caposubnets := []infrav1.Subnet{
		{
			ID:   "subnet1",
			Name: "subnet1",
			CIDR: "10.0.0.0/24",
			Tags: []string{"tag1", "tag2"},
		},
	}

	filteredSubnet := &subnets.Subnet{
		ID:   "subnet2",
		Name: "subnet2",
		CIDR: "192.168.0.0/24",
		Tags: []string{"tag3", "tag4"},
	}

	s := Service{}
	result := s.ConvertOpenStackSubnetToCAPOSubnet(caposubnets, filteredSubnet)

	expected := []infrav1.Subnet{
		{
			ID:   "subnet1",
			Name: "subnet1",
			CIDR: "10.0.0.0/24",
			Tags: []string{"tag1", "tag2"},
		},
		{
			ID:   "subnet2",
			Name: "subnet2",
			CIDR: "192.168.0.0/24",
			Tags: []string{"tag3", "tag4"},
		},
	}

	if !reflect.DeepEqual(result, expected) {
		t.Errorf("ConvertOpenStackSubnetToCAPOSubnet() = %v, want %v", result, expected)
	}
}
```